### PR TITLE
chore(KMS): cleanup isolated and redundant KMS resources

### PR DESCRIPTION
**Pull request by eeaton**
Address #1248: 
 - remove standalone CMEK configuration applied only to cai_monitoring module (tech debt from copying this module from somewhere else)
 - remove prj-$env-$bu-kms projects which are contradictory to best practice guidance on how to manage KMS host keys.  Use the pre-existing prj-$env-kms projects instead.

### CODE REVIEW

File: 0-bootstrap/main.tf [hunk 1]
```diff
@@ -60,7 +60,6 @@ module "seed_bootstrap" {
   project_prefix                 = var.project_prefix
   encrypt_gcs_bucket_tfstate     = true
   key_rotation_period            = "7776000s"
-  kms_prevent_destroy            = !var.bucket_tfstate_kms_force_destroy
```
Comment (daniel-cit) [inline 0-bootstrap/main.tf:63]: This three inputs work as a set: 

```hcl
module "kms" {
  count   = var.encrypt_gcs_bucket_tfstate ? 1 : 0 <======
  source  = "terraform-google-modules/kms/google"
  version = "~> 2.1"

  project_id           = module.seed_project.project_id
...
  key_rotation_period  = var.key_rotation_period <======
...
  encrypters = [
    "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}",
  ]
  prevent_destroy = var.kms_prevent_destroy <======
}
```
https://github.com/terraform-google-modules/terraform-google-bootstrap/blob/master/main.tf#L110C1-L115C56

Removing only the `kms_prevent_destroy` will make the build fail on the destroy of  `0-bootstrap`

**Review by daniel-cit — CHANGES_REQUESTED** [review]
Maybe regarding CMEK in `0-bootstrap`it cloud be made optional since the common KMS project does not exist yet and the user may need to enable CMEK on the Terraform state bucket

**Comment by eeaton** [comment]
You're right Daniel, I was working through some incremental changes and seeing the CI results, it has produced this error when tearing down the bootstrap stage:

> Step #34 - "destroy-bootstrap":         	            	Resource module.seed_bootstrap.module.kms[0].google_kms_crypto_key.key[0] has
Step #34 - "destroy-bootstrap":         	            	lifecycle.prevent_destroy set, but the plan calls for this resource to be
Step #34 - "destroy-bootstrap":         	            	destroyed. To avoid this error and continue with the plan, either disable
Step #34 - "destroy-bootstrap":         	            	lifecycle.prevent_destroy or reduce the scope of the plan using the -target
Step #34 - "destroy-bootstrap":         	            	option.}
Step #34 - "destroy-bootstrap":         	Test:       	TestBootstrap

I'll revise this.

And regarding _"Maybe regarding CMEK in 0-bootstrapit cloud be made optional since the common KMS project does not exist yet and the user may need to enable CMEK on the Terraform state bucket"..._
Right, it was my intent to not modify the CMEK on the bootstrap state bucket yet, but I made a mistake on the last commit. I'll be recommending we convert to [KMS Autokey](https://cloud.google.com/kms/docs/kms-with-autokey) as soon as it's GA, this will simplify the code and operation significantly from the current state. However, we'll need more design work to plan how to implement that at the bootstrap stage.

File: 0-bootstrap/main.tf (hunk 1 above)
Comment (eeaton) [reply 0-bootstrap/main.tf]: I undid this change with commit 29e0ec8

**Comment by eeaton** [comment]
All checks are green... @daniel-cit and @apeabody , can I get approval please?

**Review by daniel-cit — APPROVED** [review]
LGTM

**Comment by sleighton2022** [comment]
LGTM

**Pull request merged**